collect_app/src/main/java/org/odk/collect/android/projects/ProjectSettingsDialog.kt

```diff
@@ -110,6 +116,10 @@ class ProjectSettingsDialog : DialogFragment() {
         return binding.root
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        (requireDialog() as AlertDialog).setView(view)
```

Comment:
I ended up in a weird place here as I was using `viewLifecycleOwner` which scarily doesn't work in `onCreateDialog`. After playing with a few different options it seems like the right thing to do is to override `onCreateDialog` to create the dialog, `onCreateView` to create the view and `onViewCreated` to set the view on the dialog.

In the [docs](https://developer.android.com/guide/fragments/dialogs#custom) it suggests that we shouldn't have to do this and that just overrding `onCreateView` should do the trick. Trying this out the problem ends up being that we end up with a `Dialog` instead of an `AlertDialog` and then if we override `onCreateDialog` we use the magic of having the view automatically set on the dialog 🤯

Reply:
but you still should be able to initialize binding in `onCreateDialog()`:

```
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        binding = ProjectSettingsDialogLayoutBinding.inflate(LayoutInflater.from(context))

        return AlertDialog.Builder(requireActivity())
            .setView(binding.root)
            .create()
    }
```

then you could remove `onViewCreated()` and it would be a bit less confusing why we need to use as many lifecycle methods I think.

Reply:
Ah wait also looking at this again I've realized that we shouldn't use `viewLifecycleOwner` anyway in DialogFragments. This is in a note in the [docs](https://developer.android.com/guide/fragments/dialogs):

> Note: When subscribing to lifecycle-aware components such as LiveData, you should never use viewLifecycleOwner as the LifecycleOwner in a DialogFragment that uses Dialogs. Instead, use the DialogFragment itself, or if you're using Jetpack Navigation, use the NavBackStackEntry.

So I can definitely simplify this.